From 75516bf162985c9a5eeb7303332fa87d356d89ee Mon Sep 17 00:00:00 2001
From: Kai Meng <quic_kmeng@quicinc.com>
Date: Fri, 24 Nov 2023 16:07:03 +0800
Subject: [PATCH] policy/modules/system/systemd:Drop
 systemd_detect_virt

Fix systemd_detect_virt_t related denials, such as
type=AVC msg=audit(1651167746.131:61): avc:  denied  { nnp_transition }
for  pid=347 comm="persist-prop.sh" scontext=system_u:system_r:initrc_t:s0
tcontext=system_u:system_r:systemd_detect_virt_t:s0 tclass=process2
permissive=1

Refpolicy upstreaming master had droped systemd_detect_virt, here
just inherit from community.

Upstream-Status:Should not Upstream

Signed-off-by: Kai Meng <quic_kmeng@quicinc.com>
---
 policy/modules/system/systemd.fc | 1 -
 policy/modules/system/systemd.te | 4 ----
 2 files changed, 5 deletions(-)

diff --git a/policy/modules/system/systemd.fc b/policy/modules/system/systemd.fc
index 1a35bd65c..c0ac28bc5 100644
--- a/policy/modules/system/systemd.fc
+++ b/policy/modules/system/systemd.fc
@@ -8,7 +8,6 @@
 /usr/bin/systemd-analyze		--	gen_context(system_u:object_r:systemd_analyze_exec_t,s0)
 /usr/bin/systemd-cgtop			--	gen_context(system_u:object_r:systemd_cgtop_exec_t,s0)
 /usr/bin/systemd-coredump		--	gen_context(system_u:object_r:systemd_coredump_exec_t,s0)
-/usr/bin/systemd-detect-virt		--	gen_context(system_u:object_r:systemd_detect_virt_exec_t,s0)
 /usr/bin/systemd-hwdb			--	gen_context(system_u:object_r:systemd_hw_exec_t,s0)
 /usr/bin/systemd-nspawn			--	gen_context(system_u:object_r:systemd_nspawn_exec_t,s0)
 /usr/bin/systemd-run			--	gen_context(system_u:object_r:systemd_run_exec_t,s0)
diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 654c6a42a..df44876d2 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -102,10 +102,6 @@ init_system_domain(systemd_coredump_t, systemd_coredump_exec_t)
 type systemd_coredump_var_lib_t;
 files_type(systemd_coredump_var_lib_t)
 
-type systemd_detect_virt_t;
-type systemd_detect_virt_exec_t;
-init_daemon_domain(systemd_detect_virt_t, systemd_detect_virt_exec_t)
-
 type systemd_factory_conf_t;
 systemd_tmpfiles_conf_file(systemd_factory_conf_t)
 
-- 
2.25.1

